editor/src/clj/editor/graphics.clj

```diff
@@ -581,51 +618,77 @@
                         (catch Exception e
                           (throw (decorate-attribute-exception e attribute (first vertices))))))]
 
-                (case semantic-type
-                  :semantic-type-position
-                  (case (:coordinate-space attribute)
-                    :coordinate-space-world
-                    (let [renderable-data->world-position
-                          (case element-count
-                            3 renderable-data->world-position-v3
-                            4 renderable-data->world-position-v4)]
-                      (put-renderables! attribute-byte-offset renderable-data->world-position put-attribute-doubles!))
-                    (put-renderables! attribute-byte-offset :position-data put-attribute-doubles!))
-
-                  :semantic-type-texcoord
-                  (let [i (vswap! texcoord-index-vol inc)]
-                    (put-renderables! attribute-byte-offset #(get-in % [:texcoord-datas i :uv-data]) put-attribute-doubles!))
-
-                  :semantic-type-page-index
-                  (let [i (vswap! page-index-vol inc)]
```

Comment:
Big diff here, mostly due to indentation. We now check if `(mesh-data-exists? semantic-type channel)` to determine if we should use the attribute data from the Mesh or from the Material/Model Overrides. If there is data in the Mesh, it takes precedence for now.

Previously, the `semantic-type` determined if we should take the data from the Mesh or from the Material/Model Overrides. `:semantic-type-position`, `:semantic-type-normal`, `:semantic-type-texcoord`, and `:semantic-type-page-index` would always be taken from the Mesh, and any other attributes from the Material/Model Overrides.

This also applies to Sprites, but you get the idea. For sprites, the editor provides `:semantic-type-position`, `:semantic-type-texcoord`, and `:semantic-type-page-index` as "Mesh" data.